tap_zendesk/streams.py

```diff
@@ -139,88 +141,110 @@ def sync(self, state):
+            for rec in stream_buf:
+                yield rec
+            self.buf[stream_name] = []
+
     def sync(self, state):
         bookmark = self.get_bookmark(state)
         tickets = self.client.tickets.incremental(start_time=bookmark)
         audits_stream = TicketAudits(self.client)
-        ticket_buffer = []
-        audit_buffer = []
+        metrics_stream = TicketMetrics(self.client)
 
-        def emit_audits_metric():
-            singer.metrics.log(LOGGER, Point(metric_type='counter',
-                                             metric=singer.metrics.Metric.record_count,
-                                             value=audits_stream.count,
-                                             tags={'endpoint':audits_stream.stream.tap_stream_id}))
-            audits_stream.count = 0
+        def emit_sub_stream_metrics(sub_stream):
+            if sub_stream.is_selected():
+                singer.metrics.log(LOGGER, Point(metric_type='counter',
+                                                 metric=singer.metrics.Metric.record_count,
+                                                 value=sub_stream.count,
+                                                 tags={'endpoint':sub_stream.stream.tap_stream_id}))
+                sub_stream.count = 0
 
         if audits_stream.is_selected():
             LOGGER.info("Syncing ticket_audits per ticket...")
 
         for ticket in tickets:
-            if utils.strptime_with_tz(ticket.updated_at) < bookmark:
+            generated_timestamp_dt = datetime.datetime.utcfromtimestamp(ticket.generated_timestamp).replace(tzinfo=pytz.UTC)
+            if generated_timestamp_dt < bookmark:
                 # NB: Skip tickets that might show up because of Zendesk behavior:
                 #   The Incremental Ticket Export endpoint also returns tickets that
                 #   were updated for reasons not related to ticket events, such as a system update or a database backfill.
                 continue
 
-            self.update_bookmark(state, ticket.updated_at)
+            self.update_bookmark(state, utils.strftime(generated_timestamp_dt))
             ticket_dict = ticket.to_dict()
             ticket_dict.pop('fields') # NB: Fields is a duplicate of custom_fields, remove before emitting
+            is_deleted = (ticket_dict["status"] == "deleted")
+            should_yield = self._buffer_record((self.stream, ticket_dict))
 
-            should_yield = self._buffer_record(ticket_buffer, (self.stream, ticket_dict))
-            if should_yield:
-                for t in ticket_buffer:
-                    yield t
-                ticket_buffer = []
-
-            if audits_stream.is_selected():
-                if ticket_dict["status"] == "deleted":
-                    LOGGER.warning("Unable to retrieve audits for deleted ticket (ID: %s), skipping...", ticket_dict["id"])
-                    continue
+            if is_deleted:
```

Comment:
This should only be printed within the `if <sub_stream>.is_selected()` block below, in case one or the other isn't selected.